src/plugins/data_source/server/plugin.ts

```diff
@@ -84,17 +123,21 @@ export class DataSourcePlugin implements Plugin<DataSourcePluginSetup, DataSourc
   private createDataSourceRouteHandlerContext = (
     dataSourceService: DataSourceServiceSetup,
     cryptographyClient: CryptographyClient,
-    logger: Logger
+    logger: Logger,
+    auditTrailPromise: Promise<AuditorFactory>
   ): IContextProvider<RequestHandler<unknown, unknown, unknown>, 'dataSource'> => {
     return (context, req) => {
       return {
         opensearch: {
           getClient: (dataSourceId: string) => {
             try {
+              const auditor = auditTrailPromise.then((auditTrail) => auditTrail.asScoped(req));
+
               return dataSourceService.getDataSourceClient(
                 dataSourceId,
                 context.core.savedObjects.client,
-                cryptographyClient
+                cryptographyClient,
+                auditor
```

Comment:
is `auditTrailPromise.then` block and guarantee `auditor` is initialized/created when calling `dataSourceService.getDataSourceClient` on line 136? Or we will need to use `await`?

Reply:
meanwhile, shall we declare `getClient` as async ?

Reply:
And how about just log it here in this getClient function instead of in dataSourceService ?

Reply:
1. no, it will not block unless we use await; 2. not need to, it already returns promise 3. make sense to me so request can be private with current implementation.-- future maybe use observable.

Reply:
I also think auditor should be pass down to the configure_client. Because we'll log more than datasourceID, we'll probably log username, query(emitted by client), error, all of those are retrieved from deeper level.

Reply:
As u mentioned, this should be separate PR/task. -- As audit query still need further investigation, l'd just keep the current position and iterate in future.